silkworm/silkrpc/commands/ots_api.cpp

```diff
@@ -174,6 +185,98 @@ boost::asio::awaitable<void> OtsRpcApi::handle_ots_getBlockDetailsByHash(const n
     co_return;
 }
 
+boost::asio::awaitable<void> OtsRpcApi::handle_ots_getBlockTransactions(const nlohmann::json& request, nlohmann::json& reply) {
+
+    auto params = request["params"];
+    if (params.size() != 3) {
+        auto error_msg = "invalid ots_getBlockTransactions params: " + params.dump();
+        SILKRPC_ERROR << error_msg << "\n";
+        reply = make_json_error(request["id"], 100, error_msg);
+        co_return;
+    }
+
+    const auto block_id = params[0].get<std::string>();
+    const auto page_number = params[1].get<unsigned long>();
+    const auto page_size = params[2].get<unsigned long>();
+
+    SILKRPC_DEBUG << "block_id: " << block_id << " page_number: " << page_number << " page_size: " << page_size << "\n";
+
+    auto tx = co_await database_->begin();
+
+    try {
+        ethdb::TransactionDatabase tx_database{*tx};
+
+        const auto block_number = co_await core::get_block_number(block_id, tx_database);
+        auto block_with_hash = co_await core::read_block_by_number(*block_cache_, tx_database, block_number);
+        const auto total_difficulty = co_await core::rawdb::read_total_difficulty(tx_database, block_with_hash.hash, block_number);
+        auto receipts = co_await core::get_receipts(tx_database, block_with_hash);
+
+        const Block extended_block{block_with_hash, total_difficulty, false};
+        auto block_size = extended_block.get_block_size();
+
+        auto transaction_count = block_with_hash.block.transactions.size(); //  - 2;
```

Comment:
Remove the comment `//  -2` because it can be misleading in the future without any other information; otherwise, explain in one comment *why* it is not necessary to count for system txs here